src/services/v2/activity-history-service.js

```diff
@@ -0,0 +1,131 @@
+export class ActivityHistoryService {
+  constructor(http, adapter, apiClient) {
+    this.http = http
+    this.adapter = adapter
+    this.baseURL = 'v4/events/graphql'
+    this.apiClient = apiClient
+  }
+
+  #getOffsetDate = () => {
+    const offSetEnd = new Date()
+    const offSetStart = new Date(
+      Date.UTC(offSetEnd.getFullYear(), offSetEnd.getMonth(), offSetEnd.getDate() - 30)
+    )
+    return { offSetEnd, offSetStart }
+  }
+
+  listActivityHistoryEvents = async ({ offset = 0, limit = 1000, search = '' }) => {
+    const { offSetEnd, offSetStart } = this.#getOffsetDate()
+
+    const payload = {
+      operatioName: 'ActivityHistory',
+      query: `query ActivityHistory ($offset: Int, $limit: Int, $begin: DateTime!, $end: DateTime!) { 
+        activityHistoryEvents( offset: $offset limit: $limit, 
+          filter: { 
+            tsRange: {
+              begin: $begin, 
+              end: $end 
+            } 
+          }, orderBy: [ts_DESC] ) { 
+          ts 
+          title 
+          comment
+          type 
+          authorName 
+          authorEmail 
+          accountId 
+        } } `,
+      variables: {
+        offset,
+        limit,
+        begin: offSetStart.toISOString(),
+        end: offSetEnd.toISOString()
+      }
+    }
+
+    if (search) {
+      payload.query = `query ActivityHistory ($offset: Int, $limit: Int, $begin: DateTime!, $end: DateTime!, $search: String) { 
+      activityHistoryEvents( offset: $offset limit: $limit, 
+        filter: { 
+          tsRange: {
+            begin: $begin, 
+            end: $end 
+          } 
+          or: [{ titleIlike: $search }, { commentIlike: $search }], 
+        }, orderBy: [ts_DESC] ) 
+        { 
+          ts 
+          title 
+          comment 
+          type 
+          authorName 
+          authorEmail 
+          accountId 
+        } } `
+      payload.variables.search = `%${search}%`
+    }
```

Comment:
```suggestion
const baseQuery = `
  query ActivityHistory(
    $offset: Int, 
    $limit: Int, 
    $begin: DateTime!, 
    $end: DateTime!
    ${search ? ', $search: String' : ''}
  ) {
    activityHistoryEvents(
      offset: $offset, 
      limit: $limit, 
      filter: { 
        tsRange: { begin: $begin, end: $end }
        ${search ? 'or: [{ titleIlike: $search }, { commentIlike: $search }],' : ''}
      }, 
      orderBy: [ts_DESC]
    ) { 
      ts
      title
      comment
      type
      authorName
      authorEmail
      accountId
    }
  }
`;

const payload = {
  operationName: 'ActivityHistory',
  query: baseQuery,
  variables: {
    offset,
    limit,
    begin: offSetStart.toISOString(),
    end: offSetEnd.toISOString(),
    ...(search ? { search: `%${search}%` } : {})
  }
};
```
faca algo assim, dessa forma evita a duplicidade de código